subsys/settings/Kconfig

```diff
@@ -132,6 +140,29 @@ config SETTINGS_NVS_SECTOR_COUNT
 	help
 	  Number of sectors used for the NVS settings area
 
+config SETTINGS_ZMS_SECTOR_SIZE_MULT
+	int "Sector size of the ZMS settings area"
+	default 1
+	depends on SETTINGS_ZMS
+	help
+	  The sector size to use for the ZMS settings area as a multiple of
+	  FLASH_ERASE_BLOCK_SIZE.
+
+config SETTINGS_ZMS_SECTOR_COUNT
+	int "Sector count of the ZMS settings area"
+	default 8
+	depends on SETTINGS_ZMS
+	help
+	  Number of sectors used for the ZMS settings area
+
+config SETTINGS_ZMS_MAX_COLLISIONS_BITS
```

Comment:
Actually the collisions starts theroretically after 2000 random keys.
It doesn't matter if keys differ by only one or 2 bytes as the hash is done across 32 bits (truncated into 28).
When we reach the maximum number of collisions an -ENOSPC is returned, just like NVS backend that returns the same error if we store more than 16K keys.

Reply:
If I understand this scheme correctly, then, How does this work in the following case?
1. You have a setting stored A
2. You have a new setting to be stored B, which results in colliding hash, so you find a new hash (collision_num++ becomes 1) and write setting B.
3. Now, application/stack deleted entry "A" (for whatever reasons)
4. Now, application requests to update value B ... now the stack would calculate wrong hash, since "collision_num" is not stored, and would then store a new entry. OR when device reboots, zms_settings will calculate a wrong hash and find nothing in the storage. 

So, to make this bulletproof, may I suggest you to store this "collision_num" to ATE itself?

Reply:
@omkar3141 regarding your case :
4- When the application requests to update value B, a new entry will be created with the collision_num=0.
In this case actually we will find a double entry with the same name. because for each write of a new entry we always update the linked list hash values.
What is missing here to make it bulletproof is to take care in the delete function of the collision hash values and update their entries to take always the first element.
Does this resolve your concerns ?

Reply:
What about the other case "OR when device reboots, zms_settings will calculate a wrong hash and find nothing in the storage."?

Reply:
Right, I think I missed the hash LL part to traverse hashes. Then it is not an issue in that case.